### DISCUSSION TITLE
Clarify when the config function runs

### DISCUSSION THREAD

**Discussion by danielo515** [wbthomason/packer.nvim#873] [Ideas]
Hello, thanks for this fantastic plugin, it's really powerful.
I have a doubt about the config keyword when it is a function.
I am not sure if it runs once when the package is installed or every time the package is loaded. To be honest, I'm still a bir confused about how neovim loads plugins. I had the impression that packer just downloads and puts them on the runtime path, then neovim loads them before loading the init configuration. For that reason, I started to think that maybe config functions are only run when the package gets installed? But looking at the common usages, I think this does not makes sense.

It will be awesome if the intended usage  of this property and when it runs (every time you open neovim VS when plugin is installed) will be awesome.

Thanks.

**Comment by primeapple** [comment]
Have a look at https://github.com/wbthomason/packer.nvim#specifying-plugins=
There it says config is run each time **after** the plugin is loaded. I think this is what you need?

**Reply by danielo515** [reply]
If it were saying that, it will be more clear, but this is what it says:

> Specifies code to run after this plugin is loaded.

The term loaded is not obvious. Loaded means first installed? On each startup? when it is loaded into the cache? To me the sentence you just wrote is much more clear

**Comment by technicalpickles** [comment]
I'm curious about this too. I am trying out nvim for the first time, and using [quick.nvim](https://github.com/albingroen/quick.nvim) as a base. One thing I noticed is that some of the `use` calls have plugins, and some don't. For example:

```lua
require('lualine').setup()

return require('packer').startup(function()
  use {
    'nvim-lualine/lualine.nvim',
    requires = {'kyazdani42/nvim-web-devicons', opt = true}
  }

  use {
    "blackCauldron7/surround.nvim",
    config = function()
      require"surround".setup {mappings_style = "surround"}
    end
  }
end)
```

I tried moving the `require('lualine').setup()` to a `config` function, but it doesn't seem to be executing (ie methods they add aren't available). I added some `print` statements, and don't seem them when calling `:messages`. I've tried trimming this down to just one or two plugins, but I still don't ever see the plugins' config called. I feel like I'm missing something 🤔
